test/modules/layers/text-layer/text-layer.spec.js

```diff
@@ -54,6 +54,199 @@ test('TextLayer - sdf', t => {
   t.end();
 });
 
+test('TextLayer - MultiIconLayer sublayer positions and offsets', t => {
+  const getName = d => d.NAME;
+  const getEyeColor = d => d.EYE_COLOR;
+
+  const aliceCoordinates2d = [1, 2];
+  const aliceCoordinates3d = [...aliceCoordinates2d, 0];
+
+  const bobCoordinates2d = [3, 4];
+  const bobCoordinates3d = [...bobCoordinates2d, 0];
+
+  const emptyOffsets = [0, 0];
+  const emptyCoordinates3d = [0, 0, 0];
+
+  const testCases = [
+    {
+      props: {
+        data: [
+          {
+            NAME: 'Alice',
+            EYE_COLOR: 'blue',
+            COORDINATES: aliceCoordinates2d
+          },
+          {
+            NAME: 'Bob',
+            EYE_COLOR: 'brown',
+            COORDINATES: bobCoordinates2d
+          }
+        ],
+        getText: getName,
+        getPosition: d => d.COORDINATES,
+        updateTriggers: {
+          getText: [getName]
+        }
+      },
+      onAfterUpdate: ({subLayer}) => {
+        const {instancePositions, instanceOffsets} = subLayer.getAttributeManager().getAttributes();
+
+        t.deepEqual(
+          instancePositions.state.startIndices,
+          [0, 'Alice'.length, ('Alice' + 'Bob').length],
+          'sublayer startIndices (pre-update)'
+        );
+
+        t.deepEqual(
+          instancePositions.value,
+          [
+            ...aliceCoordinates3d, // A
+            ...aliceCoordinates3d, // l
+            ...aliceCoordinates3d, // i
+            ...aliceCoordinates3d, // c
+            ...aliceCoordinates3d, // e
+
+            ...bobCoordinates3d, // B
+            ...bobCoordinates3d, // o
+            ...bobCoordinates3d, // b
+
+            ...emptyCoordinates3d,
+            ...emptyCoordinates3d,
+            ...emptyCoordinates3d,
+            ...emptyCoordinates3d,
+            ...emptyCoordinates3d,
+            ...emptyCoordinates3d,
+            ...emptyCoordinates3d,
+            ...emptyCoordinates3d,
+            ...emptyCoordinates3d,
+            ...emptyCoordinates3d
+          ],
+          'sublayer instancePositions (pre-update)'
+        );
+
+        t.deepEqual(
+          instanceOffsets.value,
+          [
+            -76,
+            0, // A
```

Comment:
Prettier linting wouldn't let me space this nicely 😦. I can investigate prettier ignore expressions if we want to keep this around, but didn't bother before getting some PR feedback.

Reply:
I believe you can use `// prettier-ignore` just before the array

Reply:
`instanceOffsets` does not actually capture this particular bug. I would skip testing its values because they dependend on measured size of the font and might be platform specific.

Reply:
> `instanceOffsets` does not actually capture this particular bug.

Right, sorry, I should have mentioned that in the PR description. I learned about `instanceOffsets` as part of this PR and didn't see any tests around it, so I figured it might be a helpful (although superfluous) contribution.

> I would skip testing its values because they dependend on measured size of the font and might be platform specific.

Ah, ok. Good call. Deleted in 94f5a119c